cw/waverunner.py

```diff
@@ -92,7 +94,71 @@ def _ask(self, cmd):
 
     def _ask_raw(self, cmd):
         return self._instr.ask_raw(cmd)
-
+        
+    def _get_and_print_cmd_error(self):
+        """Get command error status (get, clear) for last command. On error, displays error message."""
+        # from p.328 https://cdn.teledynelecroy.com/files/manuals/maui-remote-control-and-automation-manual.pdf
+        error_msg = ["OK.",
+            "Unrecognized command/query header.", 
+            "Illegal header path.", 
+            "Illegal number.", 
+            "Illegal number suffix.", 
+            "Unrecognized keyword.", 
+            "String error.", 
+            "GET embedded in another message.", 
+            "Arbitrary data block expected.", 
+            "Non-digit character in byte count field of arbitrary data block.", 
+            "EOI detected during definite length data block transfer.", 
+            "Extra bytes detected during definite length data block transfer"]
+        return_code = int((re.findall(r'\d+', self._ask("CMR?")))[0])
+        if return_code > 11:
+            return_msg = f"{return_code} Unkown error code"
```

Comment:
Looking at the table in the manual, the codes are numbered 1..7 and 10..13, while codes 8 and 9 are unused.

Reply:
mega - thanks for catching this!